### PR TITLE
Check for accelerate env var when doing CPU only

### PR DESCRIPTION
**Pull request by muellerzr**
# What does this PR do?

Checks if `ACCELERATE_USE_CPU` is enabled to ensure trainer parts get set properly if so (and if `--use_cpu` isn't used)

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sgugger

### CODE REVIEW

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -1718,7 +1718,7 @@ def _setup_devices(self) -> "torch.device":
                 )
             AcceleratorState._reset_state(reset_partial_state=True)
         self.distributed_state = None
-        if self.use_cpu:
+        if self.use_cpu or os.environ.get("ACCELERATE_USE_CPU", False):
```
Comment (dmsuehir) [inline src/transformers/training_args.py:1721]: I was thinking that the training args would need to set `ACCELERATE_USE_CPU` to true for use_cpu/no cuda. Is that not the case?

Comment (muellerzr) [reply src/transformers/training_args.py:1721]: Nope it is not, as we pass cpu=True to the PartialState (as seen in the line below this)

Comment (dmsuehir) [reply src/transformers/training_args.py:1721]: Thanks for explaining. I looked at the code and see now how accelerate has the shared state and global dictionary that makes this work.

**Review by sgugger — APPROVED** [review]
Thanks!

**Pull request merged**